include/preprocess/max_inscribed_ellipsoid_rounding.hpp

```diff
@@ -44,18 +45,28 @@ std::tuple<MT, VT, NT> max_inscribed_ellipsoid_rounding(Polytope &P,
         E = (E + E.transpose()) / 2.0;
         E = E + MT::Identity(d, d)*std::pow(10, -8.0); //normalize E
 
-        Eigen::LLT<MT> lltOfA(E); // compute the Cholesky decomposition of E
+        Eigen::LLT<MT> lltOfA(E.inverse()); // compute the Cholesky decomposition of E^{-1}
         L = lltOfA.matrixL();
 
-        Eigen::SelfAdjointEigenSolver <MT> eigensolver(L);
-        r = eigensolver.eigenvalues().minCoeff();
-        R = eigensolver.eigenvalues().maxCoeff();
-
-        // check the roundness of the polytope
-        if(((std::abs(R / r) <= 2.3 && iter_res.second) || iter >= 20) && iter>2){
-            break;
+        // computing eigenvalues of E
+        Spectra::DenseSymMatProd<NT> op(E);
+        Spectra::SymEigsSolver<NT, Spectra::SELECT_EIGENVALUE::BOTH_ENDS, 
+                               Spectra::DenseSymMatProd<NT>> eigs(&op, 2, std::min(std::max(10, int(d)/5), int(d)));
```

Comment:
could we add a comment on the numeric values here and how we select them?

Reply:
The values is chosen empirically. I added a comment.